### PR TITLE
First delivery

### PR DESCRIPTION
**Pull request by yoelfme**
I didn't finished the search bar :(

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
+root = true
```
Comment (timgivois) [inline .editorconfig:1]: I advise that you add this to your gitignore, this file is a personal file.

File: package.json [hunk 2]
```diff
@@ -22,15 +28,26 @@
...
     "eject": "react-scripts eject",
     "lint": "eslint ./",
     "test": "yarn lint && yarn jest",
     "jest": "react-scripts test --env=jsdom --coverage"
   },
+  "lint-staged": {
+    "*.js": [
+      "prettier --write",
+      "eslint --fix",
+      "git add"
```
Comment (timgivois) [inline package.json]: not sure if this should be in the `lint-stage`, it's not related with lint

File: src/reducers/gifs.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      } = payload;
+
+      const mappedGifs = gifs.reduce((acc, gif) => {
+        acc[gif.id] = gif;
+        return acc;
+      }, {});
+
+      return { ...state, gifs: mappedGifs, loading: true };
+    }
+    case types.LOAD_DATA_FAILED: {
```
Comment (timgivois) [inline src/reducers/gifs.js:35]: i'll add a a state to know that the load failed and present it in the component.

File: src/sagas/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { all, call, put, takeEvery } from 'redux-saga/effects';
+import gifsActions from '../actions/gifs';
+import axios from 'axios';
+
+const { LOAD_DATA_START, SEARCH } = gifsActions.types;
+
+const { loadDataFinished, loadDataFailed } = gifsActions.creators;
+
+const API_URL = 'https://api.giphy.com/v1/gifs/';
+const API_KEY = 'PCv31jAmmHGTcGvY06Tow2lcIHZPMmj6';
```
Comment (timgivois) [inline src/sagas/index.js]: add it to a `.env`

File: src/actions/gifs.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { createAction } from 'redux-actions';
+
+const LOAD_DATA_START = 'LOAD_DATA_START';
+const LOAD_DATA_FINISHED = 'LOAD_DATA_FINISHED';
+const LOAD_DATA_FAILED = 'LOAD_DATA_FAILED';
+const TOGGLE_FAVORITE = 'TOGGLE_FAVORITE';
+const SEARCH = 'SEARCH';
+
+const loadDataStart = createAction(LOAD_DATA_START);
```
Comment (Robfz) [inline src/actions/gifs.js]: Nice 👍

File: src/reducers/gifs.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      const { gif, isFavorite } = payload;
+
+      if (!isFavorite) {
+        const { favorites } = state;
+        const newFavorites = { ...favorites, [gif.id]: gif };
+
+        return { ...state, favorites: newFavorites };
+      }
+
+      const { [gif.id]: _deletedFavorite, ...favorites } = state.favorites;
```
Comment (Robfz) [inline src/reducers/gifs.js:52]: Clever destructuring.

**Comment by timgivois** [comment]
Should be closed PR

**Pull request closed without merging**